### PR TITLE
Metis graph improved for mixed-dimensional meshes

### PR DESCRIPTION
**Pull request by roystgnr**
We now add links between any boundary element and its interior neighbors.

### CODE REVIEW

File: src/partitioning/parmetis_partitioner.C [hunk 1]
```diff
@@ -410,6 +416,52 @@ void ParmetisPartitioner::build_graph (const MeshBase& mesh)
...
+  // the connectivity between them and interior elements.  We can find
+  // interior elements from boundary elements, but we need to build up
+  // a lookup map to do the reverse.
+
+  typedef LIBMESH_BEST_UNORDERED_MAP<const Elem *, const Elem *>
+    map_type;
+  map_type interior_to_boundary_map;
+
+  {
+    MeshBase::const_element_iterator       elem_it  = mesh.active_elements_begin();
```
Comment (pbauman) [inline src/partitioning/parmetis_partitioner.C]: Should these be local for ParMetis?

Comment (roystgnr) [inline src/partitioning/parmetis_partitioner.C]: I don't think so.  We might have non-local boundary elements that link to local interior elements.

Comment (pbauman) [inline src/partitioning/parmetis_partitioner.C]: Ah, OK, good point.

**Comment by roystgnr** [comment]
This seems to be working; I'm going to run some more tests on private apps and then merge this afternoon unless there are objections first.

**Pull request merged**